ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs

```diff
@@ -147,19 +150,32 @@ berkeleyAccept ioManager sock = go
           (Socket.close . fst)
           $ \(sock', addr') -> do
             associateWithIOManager ioManager (Right sock')
-            return (Accepted sock' addr', go)
+
+            -- UNIX sockets don't provide a unique endpoint for the remote
+            -- side, but the InboundGovernor/Server requires one in order to
+            -- track connections.
+            -- So to differentiate clients we use a simple counter as the
+            -- remote end's address.
+            --
+            addr'' <- case addr' of
+                           Socket.SockAddrUnix _ ->
+                               return $ Socket.SockAddrUnix (show cnt)
```

Comment:
That's a hack, but I don't see a better solution.  We certainly don't want to end up with a sum type as a `LocalAddress`.

What about adding some a prefix (`ephemeral-` or `temp-`) to make it self explanatory in log files.

Reply:
Yes, it is a hack.
I'll add `temp-` to the remote end.